springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/operationdata/AbstractOperationDataScanner.java

```diff
@@ -100,17 +101,44 @@ private Message buildMessage(OperationData operationData) {
         String modelName = this.getSchemaService().register(payloadType);
         String headerModelName = this.getSchemaService().register(operationData.getHeaders());
 
-        Schema schema = payloadType.getAnnotation(io.swagger.v3.oas.annotations.media.Schema.class);
+        /*
+         * Message information can be obtained via a @AsyncMessage annotation on the method parameter, the Payload
+         * itself or via the Swagger @Schema annotation on the Payload.
+         */
+
+        var schema = payloadType.getAnnotation(Schema.class);
         String description = schema != null ? schema.description() : null;
 
-        return Message.builder()
+        var builder = Message.builder()
                 .name(payloadType.getName())
                 .title(modelName)
                 .description(description)
                 .payload(PayloadReference.fromModelName(modelName))
                 .headers(HeaderReference.fromModelName(headerModelName))
-                .bindings(operationData.getMessageBinding())
-                .build();
-    }
+                .bindings(operationData.getMessageBinding());
+
+        // Retrieve the Message information obtained from the @AsyncMessage annotation. These values have higher priority
+        //  so if we find them, we need to override the default values.
+        Message annotationMessage = operationData.getMessage();
+        if (annotationMessage != null) {
+            builder.messageId(annotationMessage.getMessageId());
+
+            var schemaFormat = annotationMessage.getSchemaFormat() != null ? annotationMessage.getSchemaFormat() : Message.DEFAULT_SCHEMA_FORMAT;
+            builder.schemaFormat(schemaFormat);
 
+            if (annotationMessage.getDescription() != null) {
+                builder.description(annotationMessage.getDescription());
+            }
+
+            if (annotationMessage.getName() != null) {
+                builder.name(annotationMessage.getName());
+            }
+
+            if (annotationMessage.getTitle() != null) {
```

Comment:
Should we check here also for notBlank, as "" is the default value in the annotation?
Assumption: If the user does not provide something, we also do not want to set/overwrite the field

Reply:
Makes total sense. Fixed